modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -0,0 +1,262 @@
+          lp.set("pad_h", attribute_proto.ints(0));
+          lp.set("pad_w", attribute_proto.ints(1));
+      } else if (attribute_proto.has_i()) {
+          lp.set(attribute_name, attribute_proto.i());
+      } else if (attribute_proto.has_f()) {
+          lp.set(attribute_name, attribute_proto.f());
+      } else if (attribute_proto.has_s()) {
+          lp.set(attribute_name, attribute_proto.s());
+      } else if (attribute_proto.floats_size() > 0) {
+          lp.set(attribute_name, DictValue::arrayReal(
+              (float*)attribute_proto.mutable_floats(), attribute_proto.floats_size()));
+      } else if (attribute_proto.ints_size() > 0) {
+          lp.set(attribute_proto.name(), DictValue::arrayInt(
+            (int*)attribute_proto.mutable_ints(), attribute_proto.ints_size()));
+    }
+
+    if (attribute_proto.has_t() || attribute_proto.has_g() ||
+            attribute_proto.strings_size() > 0 ||
+                attribute_proto.tensors_size() > 0 ||
+                    attribute_proto.graphs_size() > 0) {
+        CV_Error(Error::StsNotImplemented, "Unexpected attribute type");
+    }
+  }
+  return lp;
+}
+
+void ONNXImporter::populateNet(Net dstNet)
+{
+    CV_Assert(model_proto.has_graph());
+
+    onnx::GraphProto graph_proto = model_proto.graph();
+    std::map<std::string, Mat> weights = getWeights(graph_proto);
+    std::map<std::string, Mat>::iterator weight;
+
+    int layersSize = graph_proto.node_size();
+    LayerParams layerParams;
+    onnx::NodeProto node_proto;
+
+    for(int i = 0; i < layersSize; i++) {
+        node_proto = graph_proto.node(i);
+        layerParams = getLayerParams(node_proto);
+        if (node_proto.has_name() && !node_proto.name().empty()) {
+            layerParams.name = node_proto.name();
+        } else {
+            std::ostringstream s;
+            s << node_proto.op_type() << "_" << i;
+            layerParams.name = s.str();
+         }
+         std::string layer_type = node_proto.op_type();
+
+         if (layer_type == "MaxPool") {
+            layerParams.type = "Pooling";
+            layerParams.set("pool", "MAX");
+        } else if (layer_type == "LRN") {
+            layerParams.type = "LRN";
+            if (layerParams.has("size")) {
+                layerParams.set("local_size", layerParams.get<int>("size"));
+                layerParams.erase("size");
+            }
+        } else if (layer_type == "Gemm") {
+            layerParams.type = "InnerProduct";
+            for (int i = 1; i < node_proto.input_size(); i++) {
+                weight = weights.find(node_proto.input(i));
+                CV_Assert(weight != weights.end());
+                layerParams.blobs.push_back(weight->second);
+            }
+            layerParams.set("num_output", layerParams.blobs[0].size[0]);
+            int numLayerWeights = node_proto.input_size() - 1;
+            if (numLayerWeights == 2) {
+                layerParams.set("bias_term", true);
```

Comment:
I think `layerParams.set("bias_term", layerParams.blobs.size() == 2);` is shorter.

Reply:
done